Ednpoints.txt

```diff
@@ -0,0 +1,132 @@
+
+/************/
+
+CHEF ENDPOINTS:
+
+GET: /chefs/:chefId - get chef's info
+PATCH: /chefs/:chefId - update chef's info
+DELETE: /chefs/:chefId - delete chef's account
+
+GET: /dishes/:chefId - get all dishes of this chef
+POST: /dishes - create a new dish of this chef
+GET: /dishes/:dishId - get a particular dish
+PATCH: /dishes/:dishId - update a particular dish of this chef
+DELETE: /dishes/:dishId - delete a particular dish of this chef
+
+GET: /notifications/:chefId - get notifications of this chef
+
+GET: /orders/:chefId?status=(new, completed, canceled) - get chef's all orders
+GET: /orders/:orderId - get a particular order
+PATCH: /orders/:orderId - update status of this order
+
+POST: /reviews - create a new comment to this dish (passing dishId and chefId)
+PATCH: /reviews/:reviewId - update chef's comment to this dish
+DELETE: /reviews/:reviewId - delete chef's comment to this dish
+
+
+/************/
+
+DELIVERY PERSON ENDPOINTS:
+
+GET: /deliveries?status=(active) - get the list of all delivery persons, sorted by distance and availability based on delivery person's current location and other factors
+
+GET: /deliveries/:deliveryId - get delivery person information
+DELETE: /deliveries/:deliveryId - delete delivery person account
+PATCH: /deliveries/:deliveryId - update delivery profile info: toggle 'ready for work' status (active / not active), update 'current location' when active, update personal info
+
+GET: /orders/:deliveryId?status=(new, completed, canceled) - get info about all orders details according to their status
+
+GET: /notifications/:deliveryId - get notifications for delivery person
+
+/************/
+
+ADMIN ENDPOINTS:
+
+GET: /users - get all users' info
+GET: /users/:userId - get user's info
+PATCH: /users/:userId - update user's info
+DELETE: /users/:userId - delete user's account
+
+GET: /chefs - get all chefs' info
+GET: /chefs/:chefId - get chef's info
+PATCH: /chefs/:chefId - update chef's info
+DELETE: /chefs/:chefId - delete chef's account
+
+GET: /deliveries - get all delivery persons' info
+GET: /deliveries/:deliveryId - get a particular delivery person
+PATCH: /deliveries/:deliveryId - update a particular delivery person
+DELETE: /deliveries/:deliveryId - delete a particular delivery person
+
+GET: /dishes - get all dishes info
+GET: /dishes/:dishId - get a particular dish
+PATCH: /dishes/:dishId - update a particular dish
+DELETE: /dishes/:dishId - delete a particular dish
+
+GET: /orders - get all orders info
+GET: /orders/:orderId - get a particular order
+PATCH: /orders/:orderId - update a particular order
+DELETE: /dishes/:orderId - delete a particular order
+
+GET: /reviews?dish=dishId&chef=chefId - get comments of a particular dish
+PATCH: /reviews/:reviewId - update a particular comment
+DELETE: /reviews/:reviewId - delete a particular comment
+
+GET: /notifications/:userId - get notifications of this chef
+POST: /notifications - create new notification
```

Comment:
manually sending notifications? for what purpose?

Reply:
For example, messages about technical work, policy updates, new features